openpype/plugins/publish/integrate_new.py

```diff
@@ -108,46 +273,44 @@ class IntegrateAssetNew(pyblish.api.InstancePlugin):
     exclude_families = ["clip"]
     db_representation_context_keys = [
         "project", "asset", "task", "subset", "version", "representation",
-        "family", "hierarchy", "task", "username"
+        "family", "hierarchy", "task", "username", "frame", "udim"
```

Comment:
These keys should be always added when are available in formatting data but `frame` and `udim` should be added only if filled template used them.
```suggestion
        "family", "hierarchy", "task", "username"
```

Reply:
Isn't that exactly what the code does? It stores only the values that are used in the template data around lines 284-290. No?

Reply:
So they don't have to be defined here which may cause accidental adding of them into context where it has nothing to do?

Reply:
> So they don't have to be defined here which may cause accidental adding of them into context where it has nothing to do?

They aren't added into the context if they aren't used in the template formatting. That's how I'm reading the code before + after. But let me look at this again fresh tomorrow - I might be missing something obvious.

Reply:
These keys should be always added to representation context no matter what was used.

Reply:
Resolving this discussion as it continued [here](https://github.com/pypeclub/OpenPype/pull/2898#discussion_r840439032).